### PR TITLE
CLIP-1774: Automate Helm charts release in GitHub Actions

### PR DESCRIPTION
**Pull request by bianchi2**
This PR adds 2 new workflows and introduces a few changes to the release scripts, namely:

* there's no filtering by Jira issue key, so all commits are picked up
* changelog is generated for each product - adding path to chart root to git log command. If there are no changes, the default "Update Helm chart version" is added to changelog
* fix getting the latest version in Changelog.md, handle 2 digits after `1`
* remove duplicate git log messages (like `Update appVersions for DC apps` which is committed every week by a bot)
* add GitHub action that will generate changelog and commit changes back (triggered in `release/*` branches only)
* renames the script to be able to all a function to get changelog (if there's a hyphen in the script name, it'll fail):

```
python3 -c 'import prepare_release; print (prepare_release.gen_changelog("bamboo", "/Users/user/projects/forks/atlassian/data-center-helm-charts"))'

['* Fix typo in readme as per jira ticket', '* Update appVersions for DC apps', '* Update Bamboo Readme file']
```

Ideally, the changelog should not require any editing or deep review after the release script generates it.

Prepare release workflow will prepare chart files for a release and raise a PR with changes. Release charts workflow will cut a new release and raise a PR with changes to index.yaml

## Checklist
- [ ] I have added unit tests
- [ ] I have applied the change to all applicable products
- [ ] I have added the change description to the `changelog.md` and `Chart.yaml` files
- [ ] (Atlassian only) I have run the E2E test (if applicable)
- [ ] (Atlassian only) Internal Bamboo CI is passing

### CODE REVIEW

File: .github/workflows/prepare-release.yaml
Comment (bianchi2) [inline .github/workflows/prepare-release.yaml:24]: Even fetch-depth = 0 does not help and git log says tags are bad revision. As a workaround, clone the repo with git, checkout to a branch, run prepare_release script and then copy src back to the repo checkout in Checkout code step.

File: src/main/scripts/prepare_release.py
Comment (bianchi2) [inline src/main/scripts/prepare_release.py:46]: Adding `-- path/to/chart/root` to pick up commits relevant for a product

File: src/main/scripts/prepare_release.py
Comment (bianchi2) [inline src/main/scripts/prepare_release.py:53]: We can change this message, but in any case having empty release notes isn't good

File: src/main/scripts/prepare_release.py
Comment (bianchi2) [inline src/main/scripts/prepare_release.py:62]: We don't need Jira keys in changelog

File: src/main/scripts/prepare_release.py
Comment (bianchi2) [inline src/main/scripts/prepare_release.py:87]: Expect double digits in the second and third number, i.e. `1.10.1` or `1.10.11`

File: src/main/scripts/prepare_release.py
Comment (bianchi2) [inline src/main/scripts/prepare_release.py:71]: Wrap each string in double quotes

File: src/main/scripts/prepare_release.py
Comment (bianchi2) [inline src/main/scripts/prepare_release.py:74]: Replace `* ` with `- "` which will end up in:

```
- "Line 1"
- "Line 2"
```

File: src/main/scripts/prepare_release.py
Comment (bianchi2) [inline src/main/scripts/prepare_release.py:63]: Remove duplicate commits, e.g. Upate appVersion for DC apps

File: .github/pull_request_template.md
Comment (nghazali) [inline .github/pull_request_template.md:8]:
```suggestion
- [ ] The E2E test is passed (use `e2e` label)
```

File: src/main/scripts/prepare_release.py
Comment (bianchi2) [inline src/main/scripts/prepare_release.py:52]: Filter out commit messages from GH actions bot

File: src/main/scripts/generate_chart_repo.sh
Comment (bianchi2) [inline src/main/scripts/generate_chart_repo.sh:42]: This makes it possible to include release notes in GitHub releases page in UI

File: src/main/scripts/generate_chart_repo.sh
Comment (bianchi2) [inline src/main/scripts/generate_chart_repo.sh:38]: 1.5.0 is the latest as of today

File: src/main/scripts/generate_chart_repo.sh
Comment (bianchi2) [inline src/main/scripts/generate_chart_repo.sh:56]: No need to set entrypoint here

File: src/main/scripts/generate_chart_repo.sh
Comment (bianchi2) [inline src/main/scripts/generate_chart_repo.sh:54]: We need the entire repo since 1.5.0 helm chart releaser expects a git repository.

File: src/main/scripts/generate_chart_repo.sh
Comment (bianchi2) [inline src/main/scripts/generate_chart_repo.sh:61]: Change path for index.yaml since we're mounting an entire repo

File: .github/workflows/release-charts.yaml
Comment (bianchi2) [inline .github/workflows/release-charts.yaml:50]: Upload another md file to each chart tgz so that chart releaser can upload release notes to GitHub releases page.

File: .github/workflows/release-charts.yaml
Comment (bianchi2) [inline .github/workflows/release-charts.yaml:99]: Here we parse  Chart.yaml of any of the products to get the version which is then used in the uploader script.

File: .github/workflows/prepare-release.yaml
Comment (bianchi2) [inline .github/workflows/prepare-release.yaml:51]: We need to make sure ArtifactHub.io will like the annotations and the new version will be successfully parsed.

File: .github/workflows/prepare-release.yaml
Comment (bianchi2) [inline .github/workflows/prepare-release.yaml:51]: Since we update versions in Chart.yaml it is a good idea to refresh docs too

File: .github/workflows/prepare-release.yaml
Comment (nanux) [inline .github/workflows/prepare-release.yaml:12]: Why this need to be defined explicitly?

File: .github/workflows/prepare-release.yaml
Comment (nanux) [reply .github/workflows/prepare-release.yaml:61]: Nice!

File: .github/workflows/release-charts.yaml
Comment (nanux) [inline .github/workflows/release-charts.yaml:4]: The expectation is that after merging the PR that was generated by the action above, the release manager will go and trigger this action manually. I think that is completely fine for now, I am just confirming.

File: .github/workflows/release-charts.yaml
Comment (nanux) [inline .github/workflows/release-charts.yaml:15]: Is the signing working without issues?

File: src/main/scripts/github_asset_uploader.sh
Comment (nanux) [inline src/main/scripts/github_asset_uploader.sh:11]:
```suggestion
  echo "Must pass release version as an argument, e.g. 1.10.2"
```

File: src/main/scripts/github_asset_uploader.sh
Comment (nanux) [inline src/main/scripts/github_asset_uploader.sh:16]:
```suggestion
  echo "GITHUB_REPOSITORY env var is not set. Using the default atlassian/data/center-helm-charts"
```

File: src/main/scripts/github_asset_uploader.sh
Comment (nanux) [inline src/main/scripts/github_asset_uploader.sh:17]:
```suggestion
  export GITHUB_REPOSITORY="atlassian/data/center-helm-charts"
```

File: src/main/scripts/prepare_release.py
Comment (nanux) [reply src/main/scripts/prepare_release.py:52]: This might become a list in future.

File: src/main/scripts/prepare_release.py
Comment (nanux) [reply src/main/scripts/prepare_release.py:65]: I think this can be hoisted up to a "constant" so it is clear where to change it in case we are changing or adding Jira ticket key.

File: src/main/scripts/prepare_release.py
Comment (nanux) [inline src/main/scripts/prepare_release.py:75]: Should it be from both sides so it works for the first one as well?

File: src/main/scripts/prepare_release.py
Comment (nanux) [inline src/main/scripts/prepare_release.py:1]: Reading this script is not the easiest. I think it is ok as it is but maybe some readability finessing would be useful. For example, the short variable naming (without `_`) makes it harder to follow the code flow and it is not fully self-documenting. It would be also good to add a comment on the top describing the general functionality and processing flow.

Nevertheless, great job mate! This will save a lot of time and avoid errors.

**Review by nanux — APPROVED** [review]

File: src/main/scripts/prepare_release.py
Comment (bianchi2) [reply src/main/scripts/prepare_release.py:1]: I agree about lack of docs and some readability. I'll make it a bit nicer.

File: src/main/scripts/prepare_release.py
Comment (bianchi2) [reply src/main/scripts/prepare_release.py:65]: @nanux Do you mean the regex itself deserves to be in a variable?

File: src/main/scripts/prepare_release.py
Comment (bianchi2) [reply src/main/scripts/prepare_release.py:75]: @nanux so the original string is `* I am a string`, and first it is transformed into `* I am a string"` and then to `- "I am a string"` to fix the yaml annotations.

File: .github/workflows/release-charts.yaml
Comment (bianchi2) [reply .github/workflows/release-charts.yaml:4]: Yes, correct. Even though we can run it on every PR to main (or if it's merge common from release branch) having more control is preferable for now. So, manual run by a release engineer.

File: .github/workflows/release-charts.yaml
Comment (bianchi2) [reply .github/workflows/release-charts.yaml]: Leftovers from my test repo. Removed

File: src/main/scripts/github_asset_uploader.sh
Comment (bianchi2) [reply src/main/scripts/github_asset_uploader.sh]: Replaced with export GITHUB_REPOSITORY="${GITHUB_REPOSITORY}"

File: src/main/scripts/prepare_release.py
Comment (nanux) [reply src/main/scripts/prepare_release.py]: Yes :)

**Comment by nanux** [comment]
Thanks @bianchi2 for the readability improvements 🥇 

🟢

File: .github/workflows/release-charts.yaml
Comment (yzha645) [inline .github/workflows/release-charts.yaml:79]: We might not need to export pub key ring to the legacy format here. As it's already exported on the second next line below.

**Review by yzha645 — APPROVED** [review]

File: .github/workflows/release-charts.yaml
Comment (bianchi2) [reply .github/workflows/release-charts.yaml:79]: Right, this export is needed if we need to verify the package. I think I left it for a possible future test case where we may want to run helm verify after signing (not implemented yet)

**Pull request merged**
